app/dot/cpp/core/actions/AuthenticationAction.java

```diff
@@ -30,14 +30,24 @@
 
 public class AuthenticationAction extends Action<Authentication> {
 
-  private static final Logger logger = LoggerFactory.getLogger(AuthenticationAction.class);
+  private static final String USER_COLLECTION = "User";
+  private final Logger logger = LoggerFactory.getLogger(getClass());
 
   @Inject private MessagesApi languageService;
   @Inject private LoginService loginService;
+  @Inject private RepositoryService repositoryService;
+
+  public void setConfiguration(Authentication authenticationConfig) {
+    this.configuration = authenticationConfig;
+  }
 
   @Override
   public CompletionStage<Result> call(Request request) {
 
+    if (!databaseInitialized()) {
+      return delegate.call(request);
```

Comment:
If the database is not initialized then all calls are authorized, it doesn't seem correct.

Reply:
It is correct. If there are no entities created, no user accounts exists, so no authentication can be done. This check was added in order to make init checks to require authentication, while only the first init of the application doens't require it.

Reply:
Well, my issue here is that all calls (e.g. accessing the partners page) will lead to an internal server error with the message "key not found: USER" before initializing the database. This can be handled, like you can be taken to the login page or something, or we can just not care about it I guess.

Reply:
I think the first thing one should do is init the database. Then check views.

Reply:
Yup, I agree, it's something that happens during setup anyway, that's why I backtracked and left it up to you.